util/patch/patch.go

```diff
@@ -279,3 +297,12 @@ func (h *Helper) calculateChanges(after client.Object) (map[string]bool, error)
 	}
 	return res, nil
 }
+
+func checkNilObject(obj client.Object) error {
+	// If you're wondering why we need reflection to do this check, see https://golang.org/doc/faq#nil_error.
+	// TODO(vincepri): Remove this check and let it panic if used improperly in a future minor release.
```

Comment:
is this TODO still relevant?

Reply:
Yeah, there are tests that check for the error to be returned if a nil object is passed in, we might need to rework those tests a little, or we can just keep this check